src/main/java/org/junitpioneer/jupiter/ReportEntryExtension.java

```diff
@@ -16,26 +16,62 @@
 import static org.junitpioneer.jupiter.ReportEntry.PublishCondition.ON_FAILURE;
 import static org.junitpioneer.jupiter.ReportEntry.PublishCondition.ON_SUCCESS;
 
+import java.lang.reflect.Method;
 import java.util.Arrays;
+import java.util.List;
 import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import java.util.stream.Stream;
 
 import org.junit.jupiter.api.extension.BeforeEachCallback;
 import org.junit.jupiter.api.extension.ExtensionConfigurationException;
 import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext;
 import org.junit.jupiter.api.extension.TestWatcher;
 
-class ReportEntryExtension implements TestWatcher, BeforeEachCallback {
+class ReportEntryExtension implements TestWatcher, BeforeEachCallback, InvocationInterceptor {
+
+	private static final ExtensionContext.Namespace NAMESPACE = ExtensionContext.Namespace
+			.create(ReportEntryExtension.class);
+
+	private static final String KEY = "ReportEntry";
 
 	@Override
 	public void beforeEach(ExtensionContext context) {
-		findAnnotations(context).forEach(ReportEntryExtension::verifyKeyValueAreNotBlank);
+		findAnnotations(context).forEach(entry -> verifyReportEntry(context, entry));
 	}
 
 	private Stream<ReportEntry> findAnnotations(ExtensionContext context) {
 		return PioneerAnnotationUtils.findAllEnclosingRepeatableAnnotations(context, ReportEntry.class);
 	}
 
+	private static void verifyReportEntry(ExtensionContext context, ReportEntry entry) {
+		verifyParameterCount(context, entry);
+		verifyKeyValueAreNotBlank(entry);
+		verifyKeyNotParameterized(entry);
+	}
+
+	private static void verifyKeyNotParameterized(ReportEntry entry) {
+		if (entry.key().matches(".*\\{([0-9])+}.*")) {
+			String message = "Report entry can not have variables in the key: { key=\"%s\" value=\"%s\" }";
+			throw new ExtensionConfigurationException(format(message, entry.key(), entry.value()));
+		}
+	}
+
+	private static void verifyParameterCount(ExtensionContext context, ReportEntry entry) {
+		Matcher matcher = Pattern.compile("\\{([0-9])+}").matcher(entry.value());
+		int variableCount = 0;
+		while (matcher.find())
+			variableCount++;
+
+		if (context.getRequiredTestMethod().getParameterCount() < variableCount) {
+			String message = "Report entry contains unresolved variable(s): { key=\"%s\" value=\"%s\" }";
+			throw new ExtensionConfigurationException(format(message, entry.key(), entry.value()));
```

Comment:
This seems that this message only works if the method has one parameter less than the ReportEntry? What if there are multiple missing?  This misconfiguration should also be mentioned in the docs, as the error messages is only clear (to me) when knowing the  context.

Reply:
Why does it seem that way? I don't see it. Could you elaborate?

Reply:
Because there is one key/value reported and an exception thrown, so no more are tested. Or do I have a mind bug?

Reply:
No, I see what you mean. This is incorrect. Will fix shortly.

Reply:
Should be fixed now.